### ISSUE TITLE
Automated build on Docker Hub

### ISSUE THREAD

**Issue by lanrat** [pretalx/pretalx#564]
## Problem you are facing

Currently in order to use pretalx, you must build the image yourself. However getting updated requires you to manually check this repository for updates and re-pull and build the docker image periodically.

## Possible Solution

I am suggesting that in addition to the [pretalx-docker](https://github.com/pretalx/pretalx-docker) which has some great scripts and documentation, that pretalx be added to the [Docker Hub](https://hub.docker.com/) and enable an [automated build](https://docs.docker.com/docker-hub/builds/link-source/) so that whenever an update is pushed to pretax the container is also updated.

## Context

This will greatly assist in on-boarding new installations as well as allow for easier updates for existing Docker user.

**Comment by MaxRink** [comment]
I can only support that. Right now, the deploymentprocess isn't that stright forward to automate as it could be

**Comment by rixx** [comment]
I'm happy to admit that Docker will never be a priority for pretalx. I'll look into automated builds, but I'll have to mention that our ansible infrastructure is much better maintained than any Dockerfiles.

**Comment by lanrat** [comment]
It shouldn't be much (or any) additional effort on your part. There is already a working `Dockerfile` in the project. All that should be required is to create an account/project on the Docker Hub and link it to this repository. I've already tested it locally and everything seems to work fine.

**Comment by rixx** [comment]
> It shouldn't be much (or any) additional effort on your part.

Yes, it will be – not much, but it's there, and currently my priorities lie elsewhere.

> There is already a working Dockerfile in the project.

Nope – the Dockerfile in this project is **not recommended for use**, as is stated in the beginning of the file. The pretalx/pretalx-docker repository should be used, and it patches the currently existing Dockerfile in plenty of ways. So there is some work to be done besides just clicking a Dockerhub button. Please stay tuned, it will happen at some point.

**Comment by MaxRink** [comment]
@rixx ive started working on an pretalx-container that follows the same scheme
https://github.com/MaxRink/pretalx/tree/dockerisation

I need to rebase it though before its any useful

**Comment by rixx** [comment]
I've tried to migrate everything docker-related to our https://github.com/pretalx/pretalx-docker repository, but I think some changes were not quite right. I'm awaiting review by @morisson (who did most of the work on pretalx dockerization). Once that repository is working completely, I can just publish it to Dockerhub.

If you'd like to make a go of it, feel free!

**Comment by rixx** [comment]
Thanks to @MaxRink we now have docker builds at https://hub.docker.com/r/pretalx/standalone!

**Comment by lanrat** [comment]
Any chance we could get the docker hub build updated?

It looks like it only made one build 4 months ago.

**Comment by rixx** [comment]
I triggered an automated build and added a rule that should build on every release tag. Maybe.

**Comment by lanrat** [comment]
@rixx, I don't think that worked. The new build on the Docker hub is tagged as `1.0.3`, however it contains the code for `0.9.0`.

Additionally, the `latest` tag on docker hub was not updated.

**Comment by MaxRink** [comment]
THe issue is that pretix gets referenced as git submodule, and that reference is still pointing towards 0.9.0
This is an artefact of the fact, that the dockerfile resides in its own repository

**Comment by lanrat** [comment]
I think that moving the Docekrfile into a sub directory of the main repo would be the cleanest, but any of these solutions should be good enough.

**Comment by rixx** [comment]
> - moving everything into the main pretalx repo

No – we came from there, but that suggested to people that the docker solution was supported in any way – which it is not.

> - referencing pretalx-docker within the main pretalx repo

I don't really see how that would help.

> - updating the reference on every new version

This is what we are going with. I triggered the update manually for now, and I'm currently in the process of building an automated build and packaging infrastructure for pretalx. Once this is done, a tagged and updated version of the pretalx-docker repo should be pushed automatically on release – or it will trigger an error.
